hnn_core/gui/gui.py

```diff
@@ -320,6 +321,9 @@ def __init__(self, theme_color="#802989",
         # load default parameters
         self.params = self.load_parameters(network_configuration)
 
+        # Number of available cores
+        self.n_cores = self._available_cores()
```

Comment:
Usually, I've been told that one shouldn't use all the cores available when doing multiprocessing ... the user should be told how many cores are available and how many of those they want to use.

Reply:
Yes agreed. I documented the GUI's behavior in issue https://github.com/jonescompneurolab/hnn-core/issues/870

Using the joblib backend the number of cores is exposed and adjustable. With the MPI backend it is not. MPI uses the total number of cores-1. I was curious if there was a reason it was designed this way.

Reply:
I thought you could adjust the number of cores in MPIBackend too. Is that not the case? I think n_cores - 1 may not be the best default ... but I'm not an MPI expert. I know it's a bad idea with joblibs

Reply:
I would just set it to 1 rather than attempt to guess the total number of cores. During the tutorial, the user can be instructed to use more cores. The less magic there is in the code, the less likely it will fail.